docs/classes.md

```diff
@@ -0,0 +1,263 @@
+# Supported Class by Pyccel
+
+Pyccel strives to provide robust support for object-oriented programming concepts commonly used by developers. In Pyccel, classes are a fundamental building block for creating structured and reusable code. This documentation outlines key features and considerations when working with classes in Pyccel.
+
+## Constructor Method
+
+-   The Constructor Method, `__init__`, is used to initialize the object's attributes.
+-   It must be named `__init__`, or the Pyccel compiler will generate an error.
+-   The first parameter of the `Constructor Method` should always be named `self`.
+-   Pyccel doesn't support Interface Constructor.
```

Comment:
The term `Interface` comes from Fortran. I'm not sure it is widely understood by our users. I suggest:
```suggestion
-   Pyccel only permits one type definition for each of the arguments of the `__init__` method. Union types or templates cannot be used.
```